docs/modules/links.md

```diff
@@ -0,0 +1,125 @@
+# Dynamic Links
+
+[Firebase Dynamic Links](https://firebase.google.com/docs/dynamic-links/) allows you to create and receive links on both Android and iOS platforms. Assuming the installation instructions have been followed, Firebase Dynamic Links is ready to go.
+
+
+All Dynamic Links operations are accessed via `firebase.links()`
+
+## Create Dynamic Links
+
+RNFirebase mimics [Firebase's REST API](https://firebase.google.com/docs/dynamic-links/rest) for Dynamic Links creation.
+The differences from the REST API are:
+1. The input for the methods is a javascript object instead of a JSON object.
+2. The response contains the URL string only.
+
+### Methods
+
+#### `createDynamicLink(parameters: Object): Promise<String>`
+
+Creates a long dynamic link.
+
+```javascript
+firebase.links().createDynamicLink({
+  dynamicLinkInfo: {
+    dynamicLinkDomain: "abc123.app.goo.gl",
+    link: "https://example.com?param1=foo&param2=bar",
+    androidInfo: {
+      androidPackageName: "com.example.android"
+    },
+    iosInfo: {
+      iosBundleId: "com.example.ios"
+    }
+  }
+}).
+then((url) => {
+  // ...
+});
+```
+
+#### `createShortDynamicLink(parameters: Object): Promise<String>`
+
+Creates a short dynamic link.
+
+```javascript
+firebase.links().createShortDynamicLink(dynamicLinkInfo: {
+    dynamicLinkDomain: "abc123.app.goo.gl",
+    link: "https://example.com?param1=foo&param2=bar",
+    androidInfo: {
+      androidPackageName: "com.example.android"
+    },
+    iosInfo: {
+      iosBundleId: "com.example.ios"
+    }
+  }
+}).
+then((url) => {
+  // ...
+});
+```
+### Parameters
+
+Only the following parameters are currently supported:
+
+```javascript
+{
+  dynamicLinkInfo: {
```

Comment:
Do you think that it's necessary to break suffix out of the main info object?  I'm just thinking that it might be simpler from a user implementation point of view to not need the `dynamicLinkInfo` key at all, and just have the object as:

```
{
  dynamicLinkDomain,
  link,
  androidInfo,
  iosInfo,
  socialMetaTagInfo,
  suffix,
}
```

Reply:
I realise that this goes against the REST API, but in the interest of keeping it simpler for the user to implement it seems like an unnecessary separation.

The only point where it would make sense to keep the separation is if we were to support the long to short link conversion, which I can't see being necessary on a mobile app.

Reply:
I have change the API to be that way.